invenio_accounts/profiles/dicts.py

```diff
@@ -20,18 +20,19 @@ class ValidatedDict(dict):
     def __init__(self, schema, *args, **kwargs):
         """Constructor, validates the given data."""
         self._schema = schema() if isclass(schema) else schema
-        self._validate(dict(*args, **kwargs))
-        super().__init__(*args, **kwargs)
+        data = self._validate(dict(*args, **kwargs))
+        super().__init__(**data)
 
     def _validate(self, data):
-        """Validate the data with the dictionary's schema."""
+        """Validate the data with the dictionary's schema and return the valid value."""
         try:
             if self._schema is not None:
                 # without schema, we basically revert to a normal dictionary
                 # with more overhead
-                self._schema.load(data)
+                return self._schema.load(data)
         except ValidationError as error:
             raise ValueError(f"Validation failed: {error}")
+        return data
```

Comment:
is it needed? the code should probably not reach here

Reply:
if schema is None it acts as a normal dict...

Reply:
in this case, minor for readability:
```suggestion
            def _validate(self, data):
        """Validate the data with the dictionary's schema and return the valid value."""
        if self._schema is None:
            return data
        try:
              # without schema, we basically revert to a normal dictionary
              # with more overhead
              return self._schema.load(data)
        except ValidationError as error:
            raise ValueError(f"Validation failed: {error}")
```